<a href="https://colab.research.google.com/github/TreborRP30/An-lisis-inmobiliario-de-lima-2019/blob/main/Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import kagglehub
# Importación masiva de librerías esenciales
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

# Configuración visual para tus gráficos (para que se vean profesionales)
sns.set_theme(style="whitegrid")
plt.rcParams['figure.figsize'] = (10, 6)

# Download latest version
path = kagglehub.dataset_download("pbenavides/house-pricing-lima-peru")
print(f"Ruta descargada: {path}")

Using Colab cache for faster access to the 'house-pricing-lima-peru' dataset.
Ruta descargada: /kaggle/input/house-pricing-lima-peru


In [ ]:
# lectura
ruta = "/content/drive/MyDrive/urbania_data (1).csv"
df = pd.read_csv(ruta)
print(df.shape)
df.head()

(7663, 96)


,Unnamed: 0.1,Unnamed: 0,Antiguedad,Anunciante,Balneario,NroBanios,Nro_pisos,Cocheras,Descripcion,Direccion,...,'Aire acondicionado','Amoblado','Sauna','Calefaccion','Equipado','Terma','Portero electrico','Cocina con reposteros',Distrito,Provincia
0,0,3,5.0,Valentinoinmobiliaria,Nulo,4.0,2.0,3.0,excelente opcion para vivir en una de las mej...,"Malecon Monte Bello 5 La Molina, La Planicie, ...",...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,LaMolina,Lima
1,1,4,5.0,No disponible,Nulo,5.0,3.0,1.0,hermosa casa a espalda de futura avenida prol...,Mz. O2 Lt. 03 Urb. San Antonio De Carapongo 2 ...,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,Lurigancho,Lima
2,2,5,3.0,No disponible,Nulo,5.0,3.0,4.0,casa de estreno acabados de primera. calle ce...,"El Pinar 120 , La Planicie 1era Etapa La Molin...",...,1.0,1.0,1.0,0.0,1.0,1.0,1.0,1.0,LaMolina,Lima
3,5,8,1.0,No disponible,Sol Y Mar,5.0,2.0,3.0,linda casa en condominio con piscina club hou...,"Km 98.5 De La Panamericana Sur Asia, Sol Y Mar...",...,0.0,1.0,0.0,0.0,0.0,0.0,0.0,1.0,Asia,Lima
4,6,9,4.0,No disponible,Nulo,4.0,2.0,2.0,monterrico 2 casas en 1 ( casa bajos 4 dormit...,"Mariano De Rivera Y Ustariz #215 Surco, Huerto...",...,0.0,0.0,0.0,0.0,0.0,1.0,0.0,1.0,SantiagoDeSurco,Lima


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7663 entries, 0 to 7662
Data columns (total 96 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Unnamed: 0.1                    7663 non-null   int64  
 1   Unnamed: 0                      7663 non-null   int64  
 2   Antiguedad                      7663 non-null   float64
 3   Anunciante                      7663 non-null   object 
 4   Balneario                       7663 non-null   object 
 5   NroBanios                       7663 non-null   float64
 6   Nro_pisos                       7663 non-null   float64
 7   Cocheras                        7663 non-null   float64
 8   Descripcion                     7663 non-null   object 
 9   Direccion                       7663 non-null   object 
 10  Dormitorios                     7663 non-null   object 
 11  Estado de Inmueble              7663 non-null   object 
 12  Fecha_pub                       76

Realizó un análisis exploratorio de mi dataset para saber su estado, y lo que tomo en cuenta es:

•La cantidad de columnas y filas: 7663 columnas y 96 filas.

•Los tipos de datos de las columnas: 8 decimales, 3 enteros y 85 categoricas.

•Si hay celdas vacias o valor nulos, en mi caso no hay.

Igualmente eliminare todos las filas con un dato faltantes o nulo, en mi caso se puede permitir por el tamaño de mi dataset.

In [ ]:
df.dropna(inplace=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7663 entries, 0 to 7662
Data columns (total 96 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Unnamed: 0.1                    7663 non-null   int64  
 1   Unnamed: 0                      7663 non-null   int64  
 2   Antiguedad                      7663 non-null   float64
 3   Anunciante                      7663 non-null   object 
 4   Balneario                       7663 non-null   object 
 5   NroBanios                       7663 non-null   float64
 6   Nro_pisos                       7663 non-null   float64
 7   Cocheras                        7663 non-null   float64
 8   Descripcion                     7663 non-null   object 
 9   Direccion                       7663 non-null   object 
 10  Dormitorios                     7663 non-null   object 
 11  Estado de Inmueble              7663 non-null   object 
 12  Fecha_pub                       76

In [ ]:
# 1. ELIMINAR columnas basura
columnas_basura = ['match', 'Anunciante', 'Direccion', 'Descripcion', 'Area_constr', 'Area_total']
df.drop(columns=columnas_basura, inplace=True, errors='ignore')

# 2. RENOMBRAR las mal tipeadas
renombres = {
    'anx81tico': 'Atico',
    'Jardanxadn Interno': 'Jardin_interno',
    'Lavanderanxada': 'Lavanderia',
    'Guardiananxada': 'Guardiania',
    'Guarderanxada': 'Guarderia',
    'Jardanxadn': 'Jardin',
    'anx81reas verdes': 'Areas_verdes',
    'Areadeportiva': 'Area_deportiva',
    'Areade BBQ': 'Area_BBQ',
    'Areade sauna': 'Area_sauna',
    'Desagaue': 'Desague',
    'NroBanios': 'Nro_banios',
    'Juegos para ninios': 'Juegos_ninos'
}
df.rename(columns=renombres, inplace=True)

# 3. Formatear todas las columnas restantes (quitar espacios y comillas sueltas)
# Esto convierte "Cuarto de servicio" en "Cuarto_de_servicio" automáticamente
df.columns = df.columns.str.replace("'", "").str.replace(" ", "_")

# Verificamos el resultado
print(df.info())